### PR TITLE
Update draft-js for version 0.10.0

### PR DESCRIPTION
**Pull request by benshope**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://draftjs.org/docs/v0-10-api-migration.html#content
I just fixed the things that I needed to fix.  There could be more changes.  That link seems to describe the extent of the changes in 0.10.0.
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

One more note.  Some of the '?'s I added are based on example code from the draft-js repo not-working.  I would guess they are just mistakes in the original types - and that they are not specific to 0.10.0

### CODE REVIEW

**Comment by dmitryrogozhny** [comment]
Hi @benshope 
Thanks for the definition updates.

May I ask you, why you've changed `key` in the `EntityRange` from `number` to `string`?

It looks like it should be a `number`, as defined here - https://github.com/facebook/draft-js/blob/0.10-stable/src/model/encoding/EntityRange.js. The rest of changes looks fine for me.

It would be great, if you could update the `draft-js-tests.tsx` test file to include repo examples that you mentioned. The ones that failed for you with the current definitions.
This way you could ensure that definitions will stay up-to-date in the future, and it would help the community.

**Comment by mhegazy** [comment]
@benshope can you please update this PR.

**Comment by benshope** [comment]
Yep!  I'll update this tomorrow, thanks

**Comment by paulvanbrenk** [comment]
@benshope can you fix the merge issues?

File: draft-js/index.d.ts [hunk 1]
```diff
@@ -357,7 +357,11 @@ declare namespace Draft {
              *   - "props": Props to be passed into the React component that will be used.
              */
             interface DraftDecorator {
-                strategy: (block: ContentBlock, callback: (start: number, end: number) => void) => void;
+                strategy: (
+                    block: ContentBlock,
+                    callback: (start: number, end: number) => void,
+                    contentState?: ContentState
```
Comment (andy-ms) [inline draft-js/index.d.ts]: Is this parameter really optional? http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html

**Comment by andy-ms** [comment]
We've recently updated the repository structure.
All packages are now in a `types` subdirectory.
Please rebase this pull request onto the latest `master` branch.

**Comment by RyanCavanaugh** [comment]
Closing due to inactivity.

**Pull request closed without merging**